extensions/amp-analytics/0.1/events.js

```diff
@@ -1142,11 +1142,26 @@ export class VisibilityTracker extends EventTracker {
     const visibilitySpec = config['visibilitySpec'] || {};
     const selector = config['selector'] || visibilitySpec['selector'];
     const waitForSpec = visibilitySpec['waitFor'];
+    let reportWhenSpec = visibilitySpec['reportWhen'];
     const visibilityManager = this.root.getVisibilityManager();
     // special polyfill for eventType: 'hidden'
-    let createReadyReportPromiseFunc = null;
+    let createReportReadyPromiseFunc = null;
     if (eventType == 'hidden') {
-      createReadyReportPromiseFunc = this.createReportReadyPromise_.bind(this);
+      if (reportWhenSpec) {
+        user().error(TAG,
+            'ReportWhen should not be defined when eventType is "hidden"');
+      }
+      reportWhenSpec = eventType;
+    }
+    if (reportWhenSpec == 'hidden') {
```

Comment:
to be specific, how about `documentHidden`

Reply:
Done